File: src/main/java/forpleuvoir/hiirosakura/client/util/ServerInfoUtil.java
package forpleuvoir.hiirosakura.client.util;


import org.jetbrains.annotations.Nullable;

/**
 * 服务器信息工具类
 *
 * @author forpleuvoir
 * <p>#project_name hiirosakura
 * <p>#package forpleuvoir.hiirosakura.client.util
 * <p>#class_name ServerInfoUtil
 * <p>#create_time 2021-07-26 15:41
 */
public class ServerInfoUtil {
    @Nullable
    private static String name;
    @Nullable
    private static String address;
    @Nullable
    private static String lastServerName;
    @Nullable
    private static String lastServerAddress;
    private static int disConnectCounter = 0;

    public static void clear() {
        ServerInfoUtil.name = null;
        ServerInfoUtil.address = null;
    }

    public static void setValue(String name, String address) {
        ServerInfoUtil.name = name;
        ServerInfoUtil.address = address;
        ServerInfoUtil.lastServerName = name;
        ServerInfoUtil.lastServerAddress = address;
        ServerInfoUtil.disConnectCounter = 0;
    }

    public static void disconnect() {
        disConnectCounter++;
    }

    public static int getDisConnectCounter() {
        return disConnectCounter;
    }

    @Nullable
    public static String getName() {
        return name;
    }

    @Nullable
    public static String getAddress() {
        return address;
    }

    @Nullable
    public static String getLastServerName() {
        return lastServerName;
    }

    @Nullable
    public static String getLastServerAddress() {
        return lastServerAddress;
    }
}

File: src/main/kotlin/forpleuvoir/hiirosakura/client/util/ServerInfoUtil.kt
package forpleuvoir.hiirosakura.client.util


/**
 * 服务器信息工具类
 *
 * @author forpleuvoir
 *
 * 项目名 hiirosakura
 *
 * 包名 forpleuvoir.hiirosakura.client.util
 *
 * 文件名 ServerInfoUtil
 *
 * 创建时间 2021-07-26 15:41
 */
object ServerInfoUtil {
	@JvmStatic
	var name: String? = null
		private set
	@JvmStatic
	var address: String? = null
		private set
	@JvmStatic
	var lastServerName: String? = null
		private set
	@JvmStatic
	var lastServerAddress: String? = null
		private set
	var disConnectCounter = 0
		private set

	@JvmStatic
	fun clear() {
		name = null
		address = null
	}

	@JvmStatic
	fun setValue(name: String?, address: String?) {
		ServerInfoUtil.name = name
		ServerInfoUtil.address = address
		lastServerName = name
		lastServerAddress = address
		disConnectCounter = 0
	}

	fun disconnect() {
		disConnectCounter++
	}
}